pkg/controller/component/utils.go

```diff
@@ -0,0 +1,78 @@
+package component
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/openshift/api/image/docker10"
+	imagev1 "github.com/openshift/api/image/v1"
+	corev1 "k8s.io/api/core/v1"
+	"strconv"
+	"strings"
+)
+
+// getExposedPortsFromISI parse ImageStreamImage definition and return all exposed ports in form of ContainerPorts structs
+// borrowed from https://github.com/openshift/odo/blob/2b98a06e811a60a08c9a6545c9698896d91e8f0b/pkg/occlient/occlient.go#L541
+func getExposedPortsFromISI(image *imagev1.ImageStreamImage) ([]corev1.ContainerPort, error) {
```

Comment:
Could you use `getExposedPortsFromImageStreamImage` ?

Reply:
Done in 8a7d34a